core/runtime.rs

```diff
@@ -619,13 +673,14 @@ impl<const FOR_SNAPSHOT: bool> JsRuntimeImpl<FOR_SNAPSHOT> {
 
     drop(context_scope);
 
-    let mut js_runtime = JsRuntimeImpl {
+    let mut js_runtime = JsRuntime {
       inner: InnerIsolateState {
-        snapshotting,
+        will_snapshot,
```

Comment:
Since we splitting `SnapshotOptions` into the incoming and outgoing `SnapshotMode`, use `SnapshotMode` here.